external nccl breaks gloo compilation

Currently, you can have a libnccl on your system, and pytorch setup favors that instead of compiling one itself from source. The code that takes care of this is:
https://github.com/pytorch/pytorch/blob/master/setup.py#L316-L318

However, this breaks gloo build because gloo gets a path to NCCL via: https://github.com/pytorch/pytorch/blob/master/torch/lib/build_libs.sh#L40

I got the error during the installation:
```
CMake Error: The source directory "/home/username/pytorch/torch/lib/gloo" does not appear to contain CMakeLists.txt.
```
Is it related to this problem?

--- a/setup.py
+++ b/setup.py
@@ -277,6 +277,13 @@ if os.getenv('PYTORCH_BINARY_BUILD') and platform.system() == 'Linux':
 cwd = os.path.dirname(os.path.abspath(__file__))
 lib_path = os.path.join(cwd, "torch", "lib")
 
+# Check if you remembered to check out submodules
+gloo_cmake = os.path.join(lib_path, "gloo", "CMakeLists.txt")
+if not os.path.exists(gloo_cmake):
+    print("Could not find {}".format(gloo_cmake))
+    print("Did you run 'git submodule update --init'?")
+    sys.exit(1)
+
 tmp_install_path = lib_path + "/tmp_install"
 include_dirs += [
     cwd,

--- a/torch/lib/build_libs.sh
+++ b/torch/lib/build_libs.sh
@@ -75,6 +75,7 @@ function build() {
               -DTHCUNN_SO_VERSION=1 \
               -DTHD_SO_VERSION=1 \
               -DNO_CUDA=$((1-$WITH_CUDA)) \
+              -DNCCL_EXTERNAL=1 \
               -DCMAKE_BUILD_TYPE=$([ $DEBUG ] && echo Debug || echo Release) \
               $2
   make install -j$(getconf _NPROCESSORS_ONLN)

